src/python/pants/engine/internals/graph.py

```diff
@@ -986,16 +996,17 @@ def create_live_and_deleted_gets(
             )
             for secondary_owner_field in secondary_owner_fields:
                 matching_files.update(
-                    *secondary_owner_field.filespec_matcher.matches(list(sources_set))
```

Comment:
This was a bugfix. See https://pantsbuild.slack.com/archives/C0D7TNJHL/p1685550973185069